### PR TITLE
Adds new component: Form.Autocomplete

### PR DESCRIPTION
**Pull request by luiztiago**
## Description

Adding the `Form/Autocomplete` new component 🥳 . 

We're using the accessible autocomplete from alphagov as a base, and doing the styling update to have it similar to the VIP design system. A few usage examples are available on [component website](https://alphagov.github.io/accessible-autocomplete/examples/)

### VIP Form/Autocomplete component

image

image

## Checklist

- [ ] This PR has good automated test coverage
- [x] The storybook for the component has been updated

## Steps to Test

Outline the steps to test and verify the PR here.

Example:

1. Pull down PR.
1. `npm run dev`.
1. Open Storybook - https://deploy-preview-127--vip-design-system-components.netlify.app/?path=/story/form-autocomplete--default
1. Check how the Autocomplete behaves similarly to the component, but with accessibility. You can use a keyboard, and navigate between options, etc.

### CODE REVIEW

**Review by rianrietveld — APPROVED** [review]
@luiztiago, @djalmaaraujo 
As good as it gets ...
Very verbose, but it works.
Only to be used in the case of many options.

File: src/system/NewForm/FormAutocomplete.js [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+		width: '100%',
+		paddingRight: '40px',
+	},
+	'& .autocomplete__input--show-all-values': {
+		paddingRight: 0,
+	},
+};
+
+const inlineStyles = {
+	borderWidth: '0px',
```
Comment (djalmaaraujo) [inline src/system/NewForm/FormAutocomplete.js]: I think 0 is enough.

File: src/system/NewForm/FormAutocomplete.js [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+			isInline,
+			forLabel,
+			options,
+			label,
+			getOptionValue,
+			onChange,
+			value,
+			showAllValues = true,
+			displayMenu = 'overlay',
+			id = 'autocomplete',
```
Comment (djalmaaraujo) [inline src/system/NewForm/FormAutocomplete.js]: Can we make `vip-autocomplete`?

File: src/system/NewForm/FormAutocomplete.js [hunk 3]
```diff
@@ -0,0 +1,174 @@
...
+				const data = options.filter(
+					option => option.label.toLowerCase().indexOf( query.toLowerCase() ) >= 0
+				);
+				populateResults( data.map( option => option.label ) );
+			},
+			[ options ]
+		);
+
+		useEffect( () => {
+			document.querySelector( '.autocomplete__input' ).setAttribute( 'aria-activedescendant', '' );
```
Comment (djalmaaraujo) [inline src/system/NewForm/FormAutocomplete.js]: Try `global.document`

File: src/system/NewForm/FormAutocomplete.js [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+		}, [] );
```
Comment (djalmaaraujo) [inline src/system/NewForm/FormAutocomplete.js:133]: is this not complaining about putting document here? if yes, try `global`

**Review by djalmaaraujo — CHANGES_REQUESTED** [review]
This is amazing @luiztiago!

Thanks a lot for this component; this will be a fantastic change and impact on the VIP apps.

Left some minor suggestions.

Ps: Can you add a basic test at least with the rendered + axe no violations?

**Review by djalmaaraujo — APPROVED** [review]

**Pull request merged**